src/NextGen/DonationForm/Blocks/DonationFormBlock/Controllers/BlockRenderController.php

```diff
@@ -44,12 +47,35 @@ public function render(array $attributes)
         $donationForm = DonationForm::find($blockAttributes->formId);
 
         $viewUrl = (new GenerateDonationFormViewRouteUrl())($donationForm->id);
+        $embedId = $blockAttributes->blockId;
+
+        if ($this->shouldDisplayDonationConfirmationReceipt($embedId)) {
+            $receiptId = give_clean($_GET['givewp-receipt-id']);
+
+            $viewUrl = (new GenerateDonationConfirmationReceiptViewRouteUrl())($receiptId);
```

Comment:
Instead of mutating the value here, should we use an `if/then/else` to set the variable value?

Reply:
resolved 86a40b2d